vhdbuilder/packer/backfill-cleanup.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -x
+
+EXPIRATION_IN_HOURS=168
+# convert to seconds so we can compare it against the "tags.now" property in the resource group metadata
+(( expirationInSecs = ${EXPIRATION_IN_HOURS} * 60 * 60 ))
+# deadline = the "date +%s" representation of the oldest age we're willing to keep
+(( deadline=$(date +%s)-${expirationInSecs%.*} ))
+
+# we will clean old sig image-definitions/versions/galleries, RGs only if 
+# we have overriden the pipeline variable SKIP_IMAGE_DELETION_BACKFILL to False
+# or we are running on production pipelines
+CLEAN_OLD_WINDOWS_RESOURCES="False"
+if [[ "${SKIP_IMAGE_DELETION_BACKFILL}" == "False" || "${DRY_RUN}" == "False" ]]; then
+  CLEAN_OLD_WINDOWS_RESOURCES="True"
+fi
+
+# attempt to clean up Windows managed images and SIG image versions created over a week ago in SIG_GALLERY_NAME (cannot be the production gallery)
+# this can be used in PR check-in pipelines together with a set SIG_GALLERY_NAME from which we'd like to free up resources
+if [[ "${MODE}" == "windowsVhdMode" && -n "${AZURE_RESOURCE_GROUP_NAME}" && "${CLEAN_OLD_WINDOWS_RESOURCES}" == "True" && "${SIG_GALLERY_NAME}" != "AKSWindows" ]]; then
+  echo "Looking for Windows managed images in ${AZURE_RESOURCE_GROUP_NAME} created over ${EXPIRATION_IN_HOURS} hours ago..."
+
+  managed_image_ids=""
+  sig_version_ids=""
+
+  # delete outdated Windows managed images and associated SIG versions (.tags.os must be "Windows")
+  images=$(az image list -g ${AZURE_RESOURCE_GROUP_NAME} | jq --arg dl $deadline '.[] | select(.tags.os == "Windows") | select(.tags.now < $dl).name' | tr -d '\"' || "")
+  for image in $images; do
```

Comment:
is it expected to always delete the last image after a for loop?

Reply:
the for-loop builds up the string of resource IDs that then get deleted by a single `az resource delete` afterwards

Reply:
Can we use `managed_image_ids += " /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/images/${image}"` 

It is difficult to identify the blank between `"${managed_image_ids}` and `/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/images/${image}"`

Same for sig_version_ids

Reply:
I think that `+` should make it to clearer that this is an append operation.